src/solver/optimisation/opt_decompte_variables_et_contraintes.cpp

```diff
@@ -143,22 +136,19 @@ void OPT_DecompteDesVariablesEtDesContraintesDuProblemeAOptimiser(PROBLEME_HEBDO
         }
     }
 
-    for (Pays = 0; Pays < problemeHebdo->NombreDePays; Pays++)
+    for (int pays = 0; pays < problemeHebdo->NombreDePays; pays++)
     {
-        char Pump = problemeHebdo->CaracteristiquesHydrauliques[Pays]->PresenceDePompageModulable;
+        char Pump = problemeHebdo->CaracteristiquesHydrauliques[pays]->PresenceDePompageModulable;
         char TurbEntreBornes
-          = problemeHebdo->CaracteristiquesHydrauliques[Pays]->TurbinageEntreBornes;
+          = problemeHebdo->CaracteristiquesHydrauliques[pays]->TurbinageEntreBornes;
         char MonitorHourlyLev
-          = problemeHebdo->CaracteristiquesHydrauliques[Pays]->SuiviNiveauHoraire;
+          = problemeHebdo->CaracteristiquesHydrauliques[pays]->SuiviNiveauHoraire;
 
         if (Pump == NON_ANTARES && TurbEntreBornes == NON_ANTARES
-            && MonitorHourlyLev == NON_ANTARES)
+            && MonitorHourlyLev == NON_ANTARES
+            && problemeHebdo->CaracteristiquesHydrauliques[pays]->PresenceDHydrauliqueModulable)
```

Comment:
Please keep `OUI_ANTARES` / `NON_ANTARES` for now. We'll remove all of them in a separate PR.